runtime/Java/src/org/antlr/v4/runtime/UnbufferedCharStream.java

```diff
@@ -82,7 +82,7 @@ public UnbufferedCharStream() {
 	/** Useful for subclasses that pull char from other than this.input. */
 	public UnbufferedCharStream(int bufferSize) {
 		n = 0;
-		data = new char[bufferSize];
+		data = new int[bufferSize];
 	}
 
 	public UnbufferedCharStream(InputStream input) {
```

Comment:
Note this ends up calling `new InputStreamReader(input)` without specifying a `Charset`. That means this logic depends on the client's `$LANG` environment variable or equivalent, which is usually not what anyone wants.